build: Add support for Android NDK r23 LTS

hebasto opened the pull request:
The new Long Term Support release of the Android NDK is [available](https://groups.google.com/g/android-ndk-announce/c/MS6Qoub0DKE/m/Zfp5Ys8eAAAJ) since 2021-08-11:

> As r23 is the new LTS, the support windows for r21 and r22 have now ended.

On master (8ae4ba481ce8f7da173bef24432729c87a36cb70), dependency build fails because it expects GNU Binutils are present in the Android NDK. In [fact](https://android.googlesource.com/platform/ndk/+/master/docs/BuildSystemMaintainers.md#binutils):
> GNU Binutils remains available up to and including r22. All binutils tools with the exception of the assembler (GAS) were removed in r23. GAS was removed in r24.

This PR switches our depends build system to llvm binutils. The usage of `llvm-ar` and `llvm-ranlib` tools effectively makes r21 the minimum supported version of NDK.

With this PR:
- building depends against NDK r23 LTS now is possible with `NO_QT=1`
- building the `qt` package in depends against NDK r23 LTS still fails:
```
Creating qmake...
...

ERROR: Cannot detect Android NDK toolchain.
Please use -android-toolchain-version to specify it.
```

The issue with the `qt` package is going to be addressed in another PR.

hebasto committed:
build: Switch to llvm buinutils for Android builds

GNU Binutils with the exception of the assembler were removed in NDK r23
LTS.

fanquake commented on depends/README.md:
```diff
@@ -136,6 +136,11 @@ options will be passed to bitcoin's configure. In this case, `--disable-wallet`.
 
 Before proceeding with an Android build one needs to get the [Android SDK](https://developer.android.com/studio) and use the "SDK Manager" tool to download the NDK and one or more "Platform packages" (these are Android versions and have a corresponding API level).
+
+The minimum supported Android NDK version is [r21](https://github.com/android/ndk/wiki/Changelog-r21).
```
Rather than adding more info here, I think this whole section should be moved to `build-android.md`, instead of existing randomly at the end of the depends README.

hebasto replied:
Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/23478#issuecomment-965542506).

hebasto committed:
doc: Move Android dependencies guide into `build-android.md`

hebasto committed:
doc: Add minimum supported Android NDK version

hebasto commented:
Updated 3873023a2e922f9aa50f0bc8adbf856214765ab4 -> 4ba492052ec09d48f8c3f391cc248340e761c7f2 ([pr23478.01](https://github.com/hebasto/bitcoin/commits/pr23478.01) -> [pr23478.02](https://github.com/hebasto/bitcoin/commits/pr23478.02)):

- addressed @fanquake's [comment](https://github.com/bitcoin/bitcoin/pull/23478#discussion_r746217730)
- updated OP

hebasto commented:
Friendly ping @icota 

:tiger2:

promag commented:
> The issue with the `qt` package is going to be addressed in another PR.

Do you know if the issue exists with 5.15.2?

hebasto commented:
> Do you know if the issue exists with 5.15.2?

I'm not sure, but Android stuff has been re-worked mush in 5.14.

fanquake reviewed (approved):
ACK 4ba492052ec09d48f8c3f391cc248340e761c7f2

hebasto commented:
@promag 
> > The issue with the `qt` package is going to be addressed in another PR.
> 
> Do you know if the issue exists with 5.15.2?

See #23677.

jhay666 commented:
36cGSAQnUd6tpoRiGyxKnDJVwwQRtGGbM7
